File: apps/pos/native-ios/JuanbertosPOS/Services/OrderService.swift
import Foundation

enum OrderService {
    static func getOrders(status: String? = nil) async throws -> [Order] {
        try await APIClient.shared.request(OrderEndpoints.getOrders(status: status))
    }

    static func getOrder(id: Int) async throws -> Order {
        try await APIClient.shared.request(OrderEndpoints.getOrder(id: id))
    }

    static func createOrder(employeeId: Int, items: [CreateOrderItem]) async throws -> Order {
        let request = CreateOrderRequest(employee_id: employeeId, items: items)
        return try await APIClient.shared.request(OrderEndpoints.createOrder(request: request))
    }

    static func updateStatus(id: Int, status: String) async throws {
        try await APIClient.shared.requestVoid(OrderEndpoints.updateStatus(id: id, status: status))
    }

    static func getKitchenOrders() async throws -> [Order] {
        try await APIClient.shared.request(OrderEndpoints.getKitchenOrders())
    }
}

File: apps/pos/native-android/app/src/main/java/com/desktopkitchen/pos/services/OrderService.kt
package com.desktopkitchen.pos.services

import com.desktopkitchen.pos.models.CreateOrderItem
import com.desktopkitchen.pos.models.CreateOrderRequest
import com.desktopkitchen.pos.models.Order
import com.desktopkitchen.pos.networking.api.OrderApi
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class OrderService @Inject constructor(
    private val orderApi: OrderApi
) {
    suspend fun getOrders(status: String? = null): List<Order> = orderApi.getOrders(status)

    suspend fun getOrder(id: Int): Order = orderApi.getOrder(id)

    suspend fun createOrder(
        employeeId: Int,
        items: List<CreateOrderItem>,
        discountType: String? = null,
        discountValue: Double? = null,
        discountReason: String? = null
    ): Order = orderApi.createOrder(
        CreateOrderRequest(
            employee_id = employeeId,
            items = items,
            discount_type = discountType,
            discount_value = discountValue,
            discount_reason = discountReason
        )
    )

    suspend fun updateStatus(id: Int, status: String) {
        orderApi.updateStatus(id, mapOf("status" to status))
    }

    suspend fun getKitchenOrders(): List<Order> = orderApi.getKitchenOrders()
}
